tutor/commands/config.py

```diff
@@ -255,9 +259,41 @@ def patches_show(context: Context, name: str) -> None:
         print(rendered)
 
 
+@click.command(name="edit", help="Edit config.yml of the current environment")
```

Comment:
Optional: What do you think about adding a `-s/--save` flag here? If supplied, then instead of reminding the user to run `tutor config save`, it would just save config automatically.

Reply:
This was something I thought of adding later based on user feedback. Now that you have pointed it out, I have taken it as the feeback and added 2 features I think are QOL improvments.

1. `-s` automatically updating the environment after a successful save. The caveat here is "successful save". More on them below.
2. `-e` to specify an editor of choice. The annoying thing about desktop systems is, installing new software somehow messes up file associations. I discovered after installing KDE that my YAML files now open in Kate Editor. Now I do `tutor config edit -e (vim|emacs|..)` to open the file in a specific editor.

**Save caveats:**
* Sometimes, when I do `:wq` in Neovim (Lazyvim config), it returns a non-zero error code. This prevents the "save" from happening.
* When the file gets opened in a GUI editor like Kate, it opens it as an independent process, instead of subprocess. So the `utils.execute()` returns as soon as the editor is launched instead of return after the editor is closed. So, the environment is updated with config before editing.

Reply:
Thank you for spelling out those caveats.

> Sometimes, when I do :wq in Neovim (Lazyvim config), it returns a non-zero error code. This prevents the "save" from happening.

Could you catch this error, and if `--save` was provided, warn the user that the editor exited non-zero and thus their config changes were not applied?

> When the file gets opened in a GUI editor like Kate, it opens it as an independent process, instead of subprocess. So the utils.execute() returns as soon as the editor is launched instead of return after the editor is closed. So, the environment is updated with config before editing.

Ah, this one is a bit worrying. Visual editors often will accept a flag that blocks the command until the editor is exited, e.g. `subl --wait` for Sublime Text. That would be useful for this particular scenario, e.g.,:

    tutor config edit -s -e 'subl --wait'

This will fail with your current code, but could be fixed by running `shlex.split` on the `-e` argument. 

All that said, with all details and nuances that these new flags make us consider, I feel that we are delaying the `tutor config edit` feature more than we need to. It would be nice to have the base command Sumac. @tecoholic  how would you feel about omitting `-s` and `-e` for now, and opening a follow-up PR to add them?

Reply:
@kdmccormick Thanks for taking another look. I agree with your assessment about the flags being a bit sub-optimal. That's the reason, I initially only considered them as future improvements.

So, I will roll back those bits and leave the bare command alone in this PR.

Reply:
Sounds good!